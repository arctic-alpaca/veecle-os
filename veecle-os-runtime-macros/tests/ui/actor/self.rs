#[veecle_os_runtime_macros::actor]
async fn macro_test_actor(&self) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

#[veecle_os_runtime_macros::actor]
async fn macro_test_actor(self) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

#[veecle_os_runtime_macros::actor]
async fn macro_test_actor(self: Self) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

fn main() {}
