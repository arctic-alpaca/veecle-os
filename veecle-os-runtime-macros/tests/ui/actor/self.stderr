error: method signature may not contain self
 --> tests/ui/actor/self.rs:2:27
  |
2 | async fn macro_test_actor(&self) -> std::convert::Infallible {
  |                           ^

error: method signature may not contain self
 --> tests/ui/actor/self.rs:7:27
  |
7 | async fn macro_test_actor(self) -> std::convert::Infallible {
  |                           ^^^^

error: method signature may not contain self
  --> tests/ui/actor/self.rs:12:27
   |
12 | async fn macro_test_actor(self: Self) -> std::convert::Infallible {
   |                           ^^^^
